### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [nicnguyen3103/ML_Ecom#1]
## Goal of this Assignment
The goal of this assignment was to introduce you to 2 main concepts in Machine Learning: 
- Data Visualization
- Exploratory Data Analysis. 

You learn how to query and clean data using pandas library in Python, make some plots which help to understand more about data with Seaborn library.

## Things you did well:

- You accomplished every questions. That's a good start !
- Excellent usage of Python function.
- You know how to use different type of Pandas indexing operators. 

## Things to work on:

- Make use of more Pandas functions. Sometimes there are simpler solutions rather than creating and applying Python function.
- You should reference more about Pandas documentations, especially about indexing operator, in order not to be confused between using different type of indexing. For instance, `.loc[]` and `[]` are the same, but `[]` seems easier to be utilized. 
- Regarding your question: **How many people have a credit card that expires in 2025?**
- `ecom['CC Exp Date'].map(expire)` already successfully returns a Series of True and False, what you do to obtain the number of True is performing the summation on this Series. Therefore, the final answer is simply `sum(ecom['CC Exp Date'].map(expire))`
- FYI, If you put your boolean indexing `ecom['CC Exp Date'].map(expire)` inside a DataFrame like this : `ecom['CC Exp Date'].map(expire)`, it will return the DataFrame records that are True.
- What you actually did is not wrong but redundant, you actually selected the records that satisfies the boolean indexing: `ecom[ecom['CC Exp Date'].map(expire)]`, then you access to the 'CC Exp Date' column with `ecom[ecom['CC Exp Date'].map(expire)]['CC Exp Date]`, then apply expire() function again, this time you already filtered your CC Exp Date before, so the columns are all True. And sum() on this Series gives the right answer :).

###  To sum up
- You should break down your code into smaller parts and print them out like as i explained :D. By this you can know your code more clearly.
- Overall, you did excellent work ! Keep practicing and these should be easier than ever :D !!
